### PR TITLE
86b1tvpke

### PR DESCRIPTION
**Pull request by LakshananS**
**Add and manage students activity log**
* Create activity ( Working hours and notes)
* Update activity ( Within 2 days they can update otherwise hide the tasks)
* Delete  ( Within 2 days they can delete otherwise hide the tasks)
* Read (Any time they can view the activity)

_Unit Testing _
All the CRUD tested by jest

### CODE REVIEW

File: server/.env [hunk 1]
```diff
@@ -1 +0,0 @@
-BASE_URL=http://localhost:3000
```
Comment (bsaranga) [inline server/.env:1]: Why was this file deleted?

File: server/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+version: '3.8'
```
Comment (bsaranga) [inline server/docker-compose.yml:1]: Why is this file being committed along with your work?

File: server/src/app/api/create/create_activity.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
```
Comment (bsaranga) [inline server/src/app/api/create/create_activity.tsx]: This should be a `.ts` file if you're not including any tsx markup.

**Review by bsaranga — CHANGES_REQUESTED** [review]
1. .tsx files should have tsx markup, if not make them .ts

2. Refactor to do the data fetching inside a react server component instead of a client component. Therefore remove api routes and bring those inside the relevant pages.

3. Create a singleton to encapsulate the Prisma client. The Prisma client should be instantiated only once in the application and it should be shared throughout the application.

File: server/.env (hunk 1 above)
Comment (LakshananS) [reply server/.env:1]: I created some dummy data and used it for testing. For this reason, I deleted that file and merged it. I thought it will effect other file. and I deleted that file before the commit but it not correctly worked.

File: server/docker-compose.yml (hunk 2 above)
Comment (LakshananS) [reply server/docker-compose.yml:1]: Okay I change this one with pull request.

File: server/src/app/api/create/create_activity.tsx (hunk 3 above)
Comment (LakshananS) [reply server/src/app/api/create/create_activity.tsx]: Noted
